official/utils/testing/scripts/presubmit.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Presubmit script that run tests and lint under local environment.
+# Make sure that tensorflow and pylint is installed.
+# usage: models >: ./official/utils/testing/scripts/presubmit.sh
+set +x
+
+# Assume the pwd is under models.
+cd official
+
+exit_code=0
+
+# Check lint:
+RC_FILE="utils/testing/pylint.rcfile"
+
+echo "===========Running lint test============"
+for file in `find . -name '*.py' ! -name '*test.py' -print`
+do
+  echo "Linting ${file}"
+  pylint --rcfile="${RC_FILE}" "${file}" || exit_code=$?
+done
+
+# More lenient for test files.
+for file in `find . -name '*test.py' -print`
+do
+  echo "Linting ${file}"
+  pylint --rcfile="${RC_FILE}" --disable=missing-docstring,protected-access "${file}" || exit_code=$?
+done
+
+# Run Python test, the python version depends on the local configured defaults.
+echo "===========Running Python test============"
+
+for test_file in `find . -name '*test.py' -print`
+do
```

Comment:
May as well run in parallel. I looks like xargs can do it; otherwise it's pretty trivial to do in python with subprocess. The reason being...

Reply:
It's worth noting that even our unit tests take a long time now. How long will this take locally? In my experience, once testing takes longer than ~5 min, people stop doing it. That said, I don't want to add a ton of logic here that has to be separately maintained from the Kokoro files. In fact, I already feel nervous about this duplication. It's easy to imagine the two sets of files diverging. I don't have a solution there, but... maybe the better way to handle this is a private wrapper around the kokoro files. That way, we can include git magic, whatever you want. Runs from the internal files, on external files.

Reply:
In total it's something like 7-8 min. (So ~15 if you do py2 and py3) I agree that likelihood of being run goes down with run time.

As far as code duplication, what if the heavy lifting is in a public script and Kokoro's build.sh just finds and runs that script? (At least for testing and linting.) I *think* this is the same as what Karmel is suggesting?

Reply:
The updated version of script will the one for all. Kokoro tests will just call it.